core/src/main/java/com/linecorp/armeria/common/util/Version.java

```diff
@@ -228,4 +228,9 @@ public String toString() {
     public boolean isClean() {
         return "clean".equals(repositoryStatus);
     }
+
+    @Override
+    public int compareTo(Version ver) {
+        return this.artifactId().compareTo(ver.artifactId());
```

Comment:
We do not need `this.`.

Reply:
If two `Version`s have the same `artifactId`, they will be treated as the same version while sorting. Of course, this is not a problem in `DocService`, because there's no two `Version`s which have equal `artifactId`. However, if there's a situation in another place that we have to sort `Version`s which have the same `artifactId` but different `artifactVersion`, that will be a problem. So I think if we want to use `Comparable`, we should use all the fields. What do you think?

Reply:
Yes. That is a good idea. Maybe we don't need all fields. I think we can do `artifactId`, then `artifactVersion`, then `longCommitHash`.

But it's up to you. If you want me to do all of the fields. That is ok. Just let me know what order you want them.

Reply:
As you said, we don't need all fields. I think just `artifactId ` and `artifactVersion` are enough. :-)